testing/smoke-ide-test/src/smokeIdeTest/groovy/org/gradle/ide/sync/fixtures/IsolatedProjectsIdeSyncFixture.groovy

```diff
@@ -57,7 +57,7 @@ class IsolatedProjectsIdeSyncFixture {
             def problem = diagnostic['problem']
             assert problem instanceof List && problem.every { it instanceof Map }
             def message = configurationCacheProblemsFixture.formatStructuredMessage(problem)
-            def trace = diagnostic['trace']
+            def trace = diagnostic['trace'][0] // See ConfigurationCacheProblemsFixture
```

Comment:
We're consistent here with CCPF https://github.com/gradle/gradle/blob/0348e5c0a15230eba9ea2845c2e87f4b7ad8da8b/testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/configurationcache/ConfigurationCacheProblemsFixture.groovy#L413.

Apparently, we can reuse `CCPF` instead of having own `IPDSF` here. 
Also, the we should support extracting the new trace types